src/lib/DpApi.js

```diff
@@ -54,12 +54,20 @@ const doRequest = (async ({ url, method = 'GET', data = {}, params, options = {}
 
   try {
     const response = await fetch(url, payload)
-    const content = await response.json()
+    const contentTypeHeader = response.headers.get('Content-Type')
+    const contentType = contentTypeHeader ? contentTypeHeader.toLowerCase() : ''
+    const content = contentType.includes('json')
+      ? await response.json()
+      : contentType.includes('text')
+      ? await response.text()
```

Comment:
Why is this fallback to `response.text()` introduced?

Reply:
If the response has a _text_ content-type, is it necessary to return the text? In Axios, it is parsed automatically. Are there cases where we need to return the text, or is it acceptable to return null instead?

Reply:
I see... as we do not know which responseType a specific route may produce, indeed we have to support text as well.